dotnet/src/Functions/Functions.OpenApi/OpenApiKernelFunctionContext.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Net.Http;
+
+namespace Microsoft.SemanticKernel.Plugins.OpenApi;
+
+/// <summary>
+/// Class with data related to an Open API <see cref="KernelFunction"/> invocation.
+/// </summary>
+public sealed class OpenApiKernelFunctionContext
+{
+    /// <summary>
+    /// Key to access the <see cref="OpenApiKernelFunctionContext"/> in the <see cref="HttpRequestMessage"/>.
+    /// </summary>
+#if NET5_0_OR_GREATER
+    public static readonly HttpRequestOptionsKey<OpenApiKernelFunctionContext> KernelFunctionContextKey = new("KernelFunctionContextKey");
+#else
+    public static readonly string KernelFunctionContextKey = "KernelFunctionContextKey";
```

Comment:
nit: `Key` part is not required in actual name:

```suggestion
    public static readonly HttpRequestOptionsKey<OpenApiKernelFunctionContext> KernelFunctionContextKey = new("KernelFunctionContext");
#else
    public static readonly string KernelFunctionContextKey = "KernelFunctionContext";
```